packages/@lwc/ssr-runtime/src/reflection.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { AriaAttrNameToPropNameMap, create, entries, hasOwnProperty, isNull } from '@lwc/shared';
+import { LightningElement } from './lightning-element';
+
+// Eventually include globals that also reflect
+const attrsToProps = AriaAttrNameToPropNameMap;
+
+export function reflectAttrToProp(
+    instance: LightningElement,
+    attrName: string,
+    attrValue: string | null
+) {
+    const reflectedPropName = attrsToProps[attrName];
+    // If the reflected property was not overriden by the instance
+    if (!hasOwnProperty.call(instance, reflectedPropName)) {
```

Comment:
```suggestion
    if (reflectedPropName && !hasOwnProperty.call(instance, reflectedPropName)) {
```
Otherwise `instance.setAttribute(undefined, 'foo')` would be reflected back as `instance.undefined === 'foo'`.